### PR TITLE
allow adding of extra settings to traefik toml config

### PR DESCRIPTION
**Pull request by daniel-garcia**
This changeset allows for specifying additional configuration to the TOML file loaded by traefik.  Alternatively, every setting would have to be exposed as a separate settings in values.yaml. The settings I need are MaxIdleConnsPerHost & InsecureSkipVerify.

### CODE REVIEW

File: stable/traefik/templates/configmap.yaml [hunk 1]
```diff
@@ -10,6 +10,7 @@ metadata:
 data:
   traefik.toml: |
     # traefik.toml
+    {{ .Values.extraConfig | indent 4 }}
```
Comment (unguiculus) [inline stable/traefik/templates/configmap.yaml]: You probably should not indent this line.

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Comment by unguiculus** [comment]
cc @krancour

**Comment by unguiculus** [comment]
@daniel-garcia Please fix your CLA.

File: stable/traefik/templates/configmap.yaml (hunk 1 above)
Comment (daniel-garcia) [reply stable/traefik/templates/configmap.yaml]: fixed

File: stable/traefik/Chart.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: traefik
-version: 1.14.5
+version: 1.14.6
```
Comment (krancour) [inline stable/traefik/Chart.yaml:3]: This change should rev the minor release, not the patch level.

**Comment by krancour** [comment]
> Alternatively, every setting would have to be exposed as a separate settings in values.yaml.

I struggle with this. I very clearly see your point. In a former life, I struggled with how to efficiently manage Nginx configuration with Puppet. Very similar situation. On the one hand you have an experience (now) that's "on rails" so-to-speak, and to use this chart, you don't need to be a Traefik expert. On the other hand-- that means there's a lot of flexibility missing and who wants to go add a new feature to the chart every time they want to use some obscure setting? Not me.

I want to know what others who have contributed frequently to this chart think about this.

Ping @c-knowles

I'm also curious what degree of precedent there already may be for this sort of thing in other charts. And if there isn't... are we establishing a good precedent or a bad one?

**Comment by unguiculus** [comment]
How about allowing the configmap to be fully configurable? Would that be an option? Here's an example:

- https://github.com/kubernetes/charts/blob/master/stable/mongodb-replicaset/values.yaml#L75-L87
- https://github.com/kubernetes/charts/blob/master/stable/mongodb-replicaset/templates/mongodb-configmap.yaml#L12

**Comment by daniel-garcia** [comment]
I like the "configmap to be fully configurable" option. The biggest drawback is that it would not likely be backwards compatible with what is available now. I'm also fine with just committing the specific changes I need (if it means it gets merged). Let me know which option you want and I'll implement it. 

@unguiculus @c-knowles @krancour

**Comment by krancour** [comment]
> I like the "configmap to be fully configurable" option. The biggest drawback is that it would not likely be backwards compatible

Yes. That would be a problem.

And wouldn't this also have some other undesirable side effects?

1. You'd have to know Traefik well to configure the chart.
1. You couldn't set options using the `--set` flag during `helm install` or `helm upgrade`.

**Comment by unguiculus** [comment]
@krancour 
Re. 1: I would expect people to read the docs. If you use some software you have to learn how it works.
Re. 2: The way it is implemented now, you are right. The referenced MongoDB chart does it better. `--set` still works there.

**Comment by daniel-garcia** [comment]
So what should be the implementation here? @krancour @unguiculus

**Comment by krancour** [comment]
> Re. 1: I would expect people to read the docs. If you use some software you have to learn how it works.

I'm not saying that's wrong, but only that it's a fundamentally different experience from what this chart currently provides... and it's a BC. Right now, I can very quickly set common options by glancing at the well-documented values.yaml and tweaking the few things that might need it. Compare this to telling a user to go RTFM to familiarize themselves with traefik.toml and they're on their own...

I'll concede that our values.yaml _could_ embed a well-documented traefik.toml with sensible defaults and all other sections commented out. This would _preserve_ the quick/easy aspect of things and cut down on the RTFM factor, but again, it's a BC. I _would_ be very wiling to explore this approach, however, in a future _major_ version of this chart.

@daniel-garcia 

> So what should be the implementation here?

Let's say that the PR as currently implemented is off the table. It seems too prone to problems-- mixing a free-for-all / "do whatever you want to the traefik.toml" mechanism with the current template for traefik.toml. To highlight an example of where this could get you into trouble-- how do you go about free-form modifying a section that's already in the config produced by the template, only by _amending_ the config? If you add a new `[web]` section, for instance, what happens? It overrides the existing `[web]` section? Or extends it?

Maybe find the answer to the above and document it clearly along with an admonishment to use this feature at your own risk?

But more importantly, as I've said, I'm coming around to the idea that this proposal could actually form the basis of the next major version of this chart. Maybe that's something you'd like to work on with me?

**Comment by daniel-garcia** [comment]
I'll redo this MR and add only the features I require.

**Pull request closed without merging**